recipe/migrations/python311.yaml

```diff
@@ -0,0 +1,33 @@
+migrator_ts: 1666686085
+__migrator:
+    migration_number: 1
+    operation: key_add
+    primary_key: python
+    ordering:
+        python:
+            - 3.6.* *_cpython
```

Comment:
Do we need to list 3.6 and 3.7 here?

Reply:
Yeah was wondering the same thing. Saw this in our PyPy migrator

https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/28885e3f7b38981651764e3dd966b3a21f5db818/recipe/migrations/pypy38.yaml#L8-L9

Also we had this in the Python 3.10 migrator ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/2008 ) ( even after we dropped Python 3.6, https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1955 )

So thinking this is needed for other reasons. Hopefully someone can clarify. Also happy to drop if I've misunderstood

Reply:
Can you link a PR where this migration has been applied locally? If that works, all is fine.

Reply:
Conda-smithy keeps trying to delete the migrator altogether (as it not merged in conda-forge-pinning). Forgetting the option to tell conda-smithy to override this behavior

Reply:
It shouldn't hurt to have superfluous entries. Pretty sure we can remove 3.6.
To remove 3.7 we need to https://github.com/conda-forge/conda-forge-pinning-feedstock/issues/2623#issuecomment-1226073265 :
> \[...\] drop 3.7 when 3.11 comes out and we start the migration in October. \[...\]

As per the comment above, this PR should remove 3.7 from the current pinnings already (because AFAIR we currently don't have code to do add+remove in a single migration).

Reply:
I've made a PR in https://github.com/conda-forge/pytest-feedstock/pull/152

(The option is `--exclusive-config-file  ../conda-forge-pinning-feedstock/recipe/conda_build_config.yaml`)

Reply:
> As per the comment above, this PR should remove 3.7 from the current pinnings already (because AFAIR we currently don't have code to do add+remove in a single migration).

(NB: If there surfaced new/other (good/important) reasons to drop it after the migration, that's fine by me. I was just reciting the comment/core decision from a few months back.)

Reply:
Ah ok was looking for a way to do drop Python 3.7 in the migration, but didn't see it. Went ahead and deleted from the pinning file directly 👍

Thanks Chris! 🙏 Yeah had tried that option, but it was still deleting it for me. Probably missing something. Maybe it matters what the path to the pinning file is? In any event, looks like the PR is doing the right thing, which is encouraging 😄

Reply:
It needs to be able to find the correct `migrations/` folder next to `conda_build_config.yaml` so perhaps that was your issue?

Reply:
Oh I see. Thanks for clarifying! 🙏

Yeah probably. Was trying to point it to `conda_build_config.yaml` installed in `base` as opposed to a feedstock, which would not be adjacent to `migrations/` 😅